### PR TITLE
Read the merge flow configuration from file

### PR DESCRIPTION
**Pull request by f-alizada**
## Changes made
- Introduce the step to read the configuration from the json file which should be presented in repository. 
- The name and the path of the cofiguration file is configurable 
- The workflow parameters changed to have config_file_path and configuration_file_branch

## Reason of changing the flow
To improve UX to onboard and create the configuration

## Configuration example
```JS
{
    "merge-flow-configurations": {
        // The key of the object is source branch
        "release/8.0.3xx": {
           // The branch to which PR should be created
            "MergeToBranch": "release/8.0.4xx",
            // extra switches similar to the https://github.com/dotnet/versions/blob/main/Maestro/subscriptions.json file
            "ExtraSwitches": "-QuietComments"
        },
        "release/8.0.4xx": {
            "MergeToBranch": "main",
            "ExtraSwitches": "-QuietComments"
        }
    }
}
```

### Workflow file example
```YML
name: Usage of Inter-branch merge workflow
on: 
  push:
    branches:
      - 'main'
      - 'releases/**'


permissions:
  contents: write
  pull-requests: write

jobs:
  check-script:
    uses: dotnet/arcade/.github/workflows/inter-branch-merge-base.yml@main
    with:
      configuration_file_path: '.config/merge-flow.json'
```

## Checks
The configuration reading step was tested here: 
- https://github.com/f-alizada/sdk/actions/runs/9496551424/job/26171197494
- https://github.com/f-alizada/sdk/actions/runs/9496473279/job/26171131885

### CODE REVIEW

File: .github/workflows/scripts/read-configuration.ps1 [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+            return $null
+        }
+
+        if($mergeFlowConfig['merge-flow-configurations'].ContainsKey($MergeFromBranch)){
+            $config = $mergeFlowConfig['merge-flow-configurations'][$MergeFromBranch]
+            Write-Host "Found Configuration"
+            Write-Host $config
+            return $config
+        }else{
+            Write-Warning "There were no configuration found in default branch for $MergeFromBranch"
```
Comment (mmitche) [inline .github/workflows/scripts/read-configuration.ps1:73]: I think this is probably not a warning, just an informational.

**Comment by mmitche** [comment]
I'm fine with the config file being read from one script, then calling the second script. To me that feels more modular.

File: .github/workflows/scripts/read-configuration.ps1 (hunk 1 above)
Comment (pavelsavara) [reply .github/workflows/scripts/read-configuration.ps1:73]: "in default branch" could be replaced with full URL similar to what we have on line 54. Maybe just not "raw" link

File: .github/workflows/scripts/read-configuration.ps1 [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        }
+    }catch{
+        Write-Warning "Failed to fetch and process configuration file from default branch"
```
Comment (pavelsavara) [inline .github/workflows/scripts/read-configuration.ps1:76]: also here we could have full URL of the attempt

Comment (f-alizada) [reply .github/workflows/scripts/read-configuration.ps1]: Added the log to output the file which is being fetched for visibility before running the actual fetch.

File: .github/workflows/scripts/read-configuration.ps1 (hunk 1 above)
Comment (f-alizada) [reply .github/workflows/scripts/read-configuration.ps1]: Severity and messages are updated. 
Added the log to show which file will be fetched for configuration.

**Pull request merged**